src/coreclr/nativeaot/Runtime/MathHelpers.cpp

```diff
@@ -11,6 +11,10 @@
 
 EXTERN_C NATIVEAOT_API uint64_t REDHAWK_CALLCONV RhpDbl2ULng(double val)
 {
+#ifdef TARGET_X86
```

Comment:
Not sure why this ifdef exists? We want the same behavior on all platforms

Reply:
This is intended to be ```defined(TARGET_X86) || defined(TARGET_AMD64)```
isn't this supposed to be? Here, we are changing behavior only for ```XARCH``` and not for other platforms. Hence, the different behavior. Let me know if this makes sense or if I am understanding it wrong.